packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts

```diff
@@ -0,0 +1,63 @@
+import {HealthCheckInterface} from '../../types';
+import fs from 'fs';
+import {promisify} from 'util';
+import {findProjectRoot} from '@react-native-community/cli-tools';
+
+const applePlatformFolders = ['ios', 'macos'];
+const xcodeEnvFile = '.xcode.env';
+
+function allAppleProjectsHaveXcodeEnv(projectRoot: string): boolean {
+  return applePlatformFolders
+    .map((folder) => xcodeEnvFileExistsFor(folder, projectRoot) ?? true)
+    .reduce((previousValue, currentValue) => previousValue && currentValue);
+}
+
+/**
+ * Checks whether the basePath+platform folder exists. If that happens, it means
+ * that the project supports that platform.
+ * Returns whether the file exists in a supported platform. It returns `null` if
+ * the platform is not supported
+ * @param platform the platform to be checked
+ * @param basePath the path to the parent of the platform
+ * @returns If the platform is supported, `true` if there is the file, `false` otherwise. If the platform is not supported, `null`
+ */
+function xcodeEnvFileExistsFor(
+  platform: string,
+  basePath: string,
+): boolean | null {
+  const platformPath = basePath + platform;
+  if (fs.existsSync(platformPath)) {
+    return fs.existsSync(platformPath + xcodeEnvFile);
+  }
+  return null;
+}
+
+export default {
+  label: '.xcode.env',
+  description: 'File to customize Xcode environment',
+  getDiagnostics: async () => {
+    const projectRoot = findProjectRoot();
```

Comment:
Somewhat unrelated (and definitely not to be fixed in this PR), I looked briefly at the options that are passed to all CLI commands and it looks like they include this information. However, it doesn't look like they are properly forwarded to the individual fixers. It's a bit silly that we have to duplicate the effort to find this info again. Do we have an issue tracking this? cc @mikehardy @thymikee

Reply:
I don't think we track this. Would make sense

Reply:
Opened #1588